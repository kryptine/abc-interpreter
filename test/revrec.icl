module revrec

// This example is like reverse, but uses records which is useful to test the
// garbage collector.

import StdClass
import StdFunc
import StdInt
import StdList

:: ADT = A | B | C | D

:: Record =
	{ field_a :: ADT
	, field_b :: !ADT
	, int_a   :: Int
	, int_b   :: !Int
	}

all_records :: [Record]
all_records = [{field_a=a,field_b=b,int_a=i,int_b=j} \\ a <- all_adts, b <- all_adts, i <- [0..4], j <- [0..4]]

all_adts :: [ADT]
all_adts = [A,B,C,D]

Start = [r.field_b \\ r <- iter 20000 reverse all_records]
