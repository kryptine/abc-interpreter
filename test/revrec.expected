[(Record A C 0 37 '~' 3.7),(Record A C 0 37 '~' 4.2),(Record A C 0 37 '%' 3.7),(Record A C 0 37 '%' 4.2),(Record A C 0 42 '~' 3.7),(Record A C 0 42 '~' 4.2),(Record A C 0 42 '%' 3.7),(Record A C 0 42 '%' 4.2),(Record A C 1 37 '~' 3.7),(Record A C 1 37 '~' 4.2),(Record A C 1 37 '%' 3.7),(Record A C 1 37 '%' 4.2),(Record A C 1 42 '~' 3.7),(Record A C 1 42 '~' 4.2),(Record A C 1 42 '%' 3.7),(Record A C 1 42 '%' 4.2),(Record A D 0 37 '~' 3.7),(Record A D 0 37 '~' 4.2),(Record A D 0 37 '%' 3.7),(Record A D 0 37 '%' 4.2),(Record A D 0 42 '~' 3.7),(Record A D 0 42 '~' 4.2),(Record A D 0 42 '%' 3.7),(Record A D 0 42 '%' 4.2),(Record A D 1 37 '~' 3.7),(Record A D 1 37 '~' 4.2),(Record A D 1 37 '%' 3.7),(Record A D 1 37 '%' 4.2),(Record A D 1 42 '~' 3.7),(Record A D 1 42 '~' 4.2),(Record A D 1 42 '%' 3.7),(Record A D 1 42 '%' 4.2),(Record B C 0 37 '~' 3.7),(Record B C 0 37 '~' 4.2),(Record B C 0 37 '%' 3.7),(Record B C 0 37 '%' 4.2),(Record B C 0 42 '~' 3.7),(Record B C 0 42 '~' 4.2),(Record B C 0 42 '%' 3.7),(Record B C 0 42 '%' 4.2),(Record B C 1 37 '~' 3.7),(Record B C 1 37 '~' 4.2),(Record B C 1 37 '%' 3.7),(Record B C 1 37 '%' 4.2),(Record B C 1 42 '~' 3.7),(Record B C 1 42 '~' 4.2),(Record B C 1 42 '%' 3.7),(Record B C 1 42 '%' 4.2),(Record B D 0 37 '~' 3.7),(Record B D 0 37 '~' 4.2),(Record B D 0 37 '%' 3.7),(Record B D 0 37 '%' 4.2),(Record B D 0 42 '~' 3.7),(Record B D 0 42 '~' 4.2),(Record B D 0 42 '%' 3.7),(Record B D 0 42 '%' 4.2),(Record B D 1 37 '~' 3.7),(Record B D 1 37 '~' 4.2),(Record B D 1 37 '%' 3.7),(Record B D 1 37 '%' 4.2),(Record B D 1 42 '~' 3.7),(Record B D 1 42 '~' 4.2),(Record B D 1 42 '%' 3.7),(Record B D 1 42 '%' 4.2)]
